pallets/transfer-allowlist/src/benchmarking.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use cfg_test_utils::system::time_travel::advance_n_blocks;
+use cfg_traits::fees::Fees;
+use codec::EncodeLike;
+use frame_benchmarking::*;
+use frame_support::traits::{Currency, Get, ReservableCurrency};
+use frame_system::RawOrigin;
+use scale_info::TypeInfo;
+use sp_runtime::traits::{AtLeast32BitUnsigned, Bounded, One};
+
+use super::*;
+
+#[cfg(test)]
+fn config_mocks() {
+	use crate::mock::Fees;
+
+	Fees::mock_fee_value(|_| 0);
+	Fees::mock_fee_to_author(|_, _| Ok(()));
+}
+
+benchmarks! {
+	where_clause {
+		where
+		<T as frame_system::Config>::AccountId: Into<<T as pallet::Config>::Location>,
+		<T as pallet::Config>::Location: From<<T as frame_system::Config>::AccountId> + EncodeLike<<T as pallet::Config>::Location>,
+			<T as pallet::Config>::ReserveCurrency: Currency<<T as frame_system::Config>::AccountId> + ReservableCurrency<<T as frame_system::Config>::AccountId>,
+		<T as pallet::Config>::CurrencyId: Default,
+		<T as frame_system::Config>::BlockNumber: AtLeast32BitUnsigned + Bounded + TypeInfo
+
+	}
+
+	add_transfer_allowance_no_existing_metadata {
+		let (sender, receiver) = set_up_users::<T>();
+	}:add_transfer_allowance(RawOrigin::Signed(sender.clone()), T::CurrencyId::default(), receiver.clone().into())
+
+
+	add_transfer_allowance_existing_metadata {
+		let (sender, receiver) = set_up_users::<T>();
+		Pallet::<T>::add_allowance_delay(RawOrigin::Signed(sender.clone()).into(), T::CurrencyId::default(), 200u32.into())?;
+	}:add_transfer_allowance(RawOrigin::Signed(sender.clone()), T::CurrencyId::default(), receiver.clone().into())
```

Comment:
Not sure which resolution to [this thread](https://github.com/centrifuge/centrifuge-chain/pull/1323#discussion_r1187061233) you've chosen @thea-leake, if finally you go without the `max()`, you can remove one of this benchmarks